packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -483,30 +478,124 @@ export class PerformanceEvent {
     }
 }
 
+// Note - these Telemetry types should move to common-definitions package
+export enum TelemetryDataTag {
+    None,
+    CodeArtifact,
```

Comment:
i don't really understand what a CodeArtifact is. Labeling like this is always hard, and i'm not sure it'll ever be robust or clear enough for consumers to take a dependency on. I think i prefer justs a boolen, basically this data might be usafe. i guess we could keep the tag, but call it something like PIITag and then the category is just a sub-category

Reply:
rather than CodeArtifact maybe Application, Developer, or Package Data, the counterpart to that would then be UserData. For the specific instance above that string is determined and controlled by the container author.

Reply:
Yeah I came up with like 4 different possible names here and don't like any of them :P  I think PackageData v. UserData is a nice idea, I may switch to that.

To respond to your first comment - I do want more granularity than boolean "hash this" v. "it's fine". e.g. PackageData can be logged plainly in our current product, but in other cases should be scrubbed.